semgrep/semgrep/constants.py

```diff
@@ -13,6 +13,13 @@
 DEFAULT_CONFIG_FOLDER = f".{DEFAULT_SEMGREP_CONFIG_NAME}"
 
 YML_EXTENSIONS = {".yml", ".yaml"}
+LANGUAGE_EXTENSIONS = {
```

Comment:
@aryx note the semi-coupling. @ulziibay could we automatically generate this mapping from the work you did in semgrep-core?

Reply:
@ievans were we talking about making language field optional? Would be nice to remove from the list of required fields and remove the extension hard-coding in semgrep-core

Reply:
`pattern-regex` is now operating on all target files and makes use of include/exclude. It's now also being composed with other patterns, so no need for language extension checks :+1: